lib/std/math/hypot.zig

```diff
@@ -1,176 +1,110 @@
-// Ported from musl, which is licensed under the MIT license:
-// https://git.musl-libc.org/cgit/musl/tree/COPYRIGHT
-//
-// https://git.musl-libc.org/cgit/musl/tree/src/math/hypotf.c
-// https://git.musl-libc.org/cgit/musl/tree/src/math/hypot.c
-
 const std = @import("../std.zig");
 const math = std.math;
 const expect = std.testing.expect;
-const maxInt = std.math.maxInt;
+const isNan = math.isNan;
+const isInf = math.isInf;
+const inf = math.inf;
+const nan = math.nan;
+const floatEpsAt = math.floatEpsAt;
+const floatEps = math.floatEps;
+const floatMin = math.floatMin;
+const floatMax = math.floatMax;
 
 /// Returns sqrt(x * x + y * y), avoiding unnecessary overflow and underflow.
 ///
 /// Special Cases:
 ///
 /// |   x   |   y   | hypot |
 /// |-------|-------|-------|
-/// | +inf  |  num  | +inf  |
-/// |  num  | +-inf | +inf  |
-/// |  nan  |  any  |  nan  |
-/// |  any  |  nan  |  nan  |
-pub fn hypot(x: anytype, y: anytype) @TypeOf(x, y) {
+/// | +-inf |  any  | +inf  |
+/// |  any  | +-inf | +inf  |
+/// |  nan  |  fin  |  nan  |
+/// |  fin  |  nan  |  nan  |
+pub fn hypot(x: anytype, y: anytype) Hypot(@TypeOf(x, y)) {
     const T = @TypeOf(x, y);
-    return switch (T) {
-        f32 => hypot32(x, y),
-        f64 => hypot64(x, y),
+    switch (@typeInfo(T)) {
+        .Float => {},
+        .ComptimeFloat => return @sqrt(x * x + y * y),
+        .ComptimeInt => {
```

Comment:
How useful do you think the integer hypot is? I feel like we remove this branch (and the return type function) and just keep the float for now and let the caller do the cast if they need it.

There is some precedence though for sqrt, log etc but I know of use-cases for integer applications here. In any case if we want integer support we should also support general ints and not just comptime.

Reply:
If this is just for ergonomics I would prefer we omit for now. I feel the change in output type for comptime_int is unexpected when compared to other math functions (e.g. integer sqrt) since we don't do this anywhere else.

I do see your point about it being unambiguous but we also tend to prefer explicitness even if its more verbose generally.